overlord/configstate/applyconfighandler.go

```diff
@@ -38,11 +35,22 @@ func (h *applyConfigHandler) Before() error {
 // Done is called by the HookManager after the apply-config hook has exited
 // successfully.
 func (h *applyConfigHandler) Done() error {
+	// Save any configurations changes the hook may have made
+	h.transaction.Commit()
 	return nil
 }
 
 // Error is called by the HookManager after the apply-config hook has exited
 // non-zero, and includes the error.
 func (h *applyConfigHandler) Error(err error) error {
+	// Note that in this case, Done() is not called, which means any
+	// configuration changes made by the hook are dropped.
 	return nil
 }
+
+// SetConf is called by `snapctl set` to associate the value with the key in
```

Comment:
Well, talking to @pedronis it sounded like such functionality might actually be desired.